asapdiscovery-modeling/asapdiscovery/modeling/tests/test_protein_prep.py

```diff
@@ -0,0 +1,227 @@
+        assert dataset != target_dataset
+
+    def test_dataset_pickle(self, target_dataset, output_dir, pkl_name="to_prep.pkl"):
+        pkl_file = output_dir / pkl_name
+        target_dataset.to_pkl(pkl_file)
+        assert pkl_file.exists()
+        assert pkl_file.is_file()
+
+        loaded_dataset = PreppedTargets.from_pkl(pkl_file)
+
+        assert loaded_dataset == target_dataset
+
+    @pytest.mark.skip(
+        reason="Multiple embedded schema objects need more logic to serialize to json"
+    )
+    def test_dataset_json(self, target_dataset, output_dir, json_name="to_prep.json"):
+        json_file = output_dir / json_name
+        target_dataset.to_json(json_file)
+        assert json_file.exists()
+        assert json_file.is_file()
+
+        loaded_dataset = PreppedTargets.from_json(json_file)
+
+        assert loaded_dataset == target_dataset
+
+
+@pytest.fixture
+def prep_dict(mers_target, sars_target):
+    return {
+        "mers": (mers_target, fetch_test_file("mpro_mers_seqres.yaml")),
+        "sars": (sars_target, fetch_test_file("mpro_sars2_seqres.yaml")),
+    }
+
+
+class TestProteinPrep:
+    @pytest.mark.parametrize("target_name", ["mers", "sars"])
+    def test_protein_prep_workflow(
+        self,
+        target_name,
+        prep_dict,
+        ref,
+        output_dir,
+        loop_db,
+        reference_output_files,
+        ref_chain="A",
+    ):
+        target, seqres_yaml = prep_dict[target_name]
+        print(target, seqres_yaml)
+        prep_opts = PrepOpts(
+            ref_fn=ref,
+            ref_chain=ref_chain,
+            loop_db=loop_db,
+            seqres_yaml=seqres_yaml,
+            output_dir=output_dir,
+        )
+        prepped_target = protein_prep_workflow(target, prep_opts)
+
+        generated_output_files = {}
+        if "protein" in prepped_target.molecule_filter.components_to_keep:
+            generated_output_files["protein"] = prepped_target.protein
+            generated_output_files["design_unit"] = prepped_target.design_unit
+        if "ligand" in prepped_target.molecule_filter.components_to_keep:
+            generated_output_files["ligand"] = prepped_target.ligand
+            generated_output_files["complex"] = prepped_target.complex
+        for component, fn in generated_output_files.items():
+            assert Path(fn).exists()
+            assert Path(fn).is_file()
+
+        # Load the prepared design unit
+        du = oechem.OEDesignUnit()
+        oechem.OEReadDesignUnit(str(prepped_target.design_unit), du)
+
+        assert type(du) == oechem.OEDesignUnit
+        assert du.HasReceptor()
+        if "ligand" in prepped_target.molecule_filter.components_to_keep:
+            assert du.HasLigand()
+        # TODO: Find a better way to test if the output matches the excepted output in
+        #  reference_output_files, as hashing does not work due to
+        #  any minor changes in atom positions causing a different hash
```

Comment:
You could use `np.assert_allclose(rtol=XXX)` to check the positions are close enough.